toolchain/sem_ir/name_scope.h

```diff
@@ -281,6 +289,10 @@ class NameScope : public Printable<NameScope> {
   // True if this is the `Cpp` namescope used when importing C++ code.
   bool is_cpp_scope_ = false;
 
+  // True if this is the scope of an interface definition, where associated
+  // entities will be bound to the interface's `Self` symbolic type.
+  bool is_interface_definition_ = false;
```

Comment:
Do we need import support for this? Or is the logic that this only matters within the definition of the interface itself, so is never relevant on import?

If the latter, I think we'd benefit from a comment explaining that. Actually, for consistency it might be nice to set the flag to false once we reach the end of the interface definition, so that the behavior is the same after the interface definition as it is in an importer, and rename it to `is_interface_being_defined` or similar.

Reply:
My concern is we want this behavior for out-of-line definitions, as in:

```
interface I {
  let T:! type;
  default fn F() -> T;
}

// This use of `T` should be bound to `I`'s `Self`.
fn I.F() -> T;
```

I will add support to import for this flag.